src/workflows/airqo_etl_utils/satellilte_utils.py

```
@@ -0,0 +1,160 @@
+            missing_columns = required_columns.difference(data.columns)
+            raise ValueError(
+                f"Provided dataframe missing necessary columns: {', '.join(missing_columns)}"
+            )
+        try:
+            data["date"] = pd.to_datetime(data["date"])
+        except ValueError as e:
+            raise ValueError(
+                "datetime conversion error, please provide timestamp in valid format"
+            )
+        group_columns = (
+            ["ID"] + additional_columns if job_type == "prediction" else ["ID"]
```

Ruff

Undefined name `additional_columns`

(F821)